gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/DistributedIdentifyUtil.java

```diff
@@ -0,0 +1,278 @@
+import org.slf4j.LoggerFactory;
+
+public class DistributedIdentifyUtil implements Serializable {
+
+  private static final long serialVersionUID = -5412153338307931076L;
+  private final SparkSession session;
+  private final Map<String, String> clientBuilderConfiguration;
+
+  public DistributedIdentifyUtil(
+      SparkSession session, Map<String, String> clientBuilderConfiguration) {
+    this.session = session;
+    this.clientBuilderConfiguration = clientBuilderConfiguration;
+  }
+
+  public static NessieApiV1 getApi(Map<String, String> configuration) {
+    // TODO: add this config - nessie.client-builder-impl
+    String clientBuilderClassName = configuration.get("nessie.client-builder-impl");
+    if (clientBuilderClassName == null) {
+      // Use default HttpClientBuilder
+      return HttpClientBuilder.builder().fromConfig(configuration::get).build(NessieApiV1.class);
+    }
+    // Use custom client builder
+    try {
+      NessieClientBuilder builder =
+          (NessieClientBuilder)
+              Class.forName(clientBuilderClassName).getDeclaredConstructor().newInstance();
+      return (NessieApiV1) builder.fromConfig(configuration::get).build(NessieApiV1.class);
+    } catch (ClassNotFoundException e) {
+      throw new IllegalArgumentException(
+          String.format("No custom client builder class found for '%s' ", clientBuilderClassName));
+    } catch (InvocationTargetException
+        | InstantiationException
+        | IllegalAccessException
+        | NoSuchMethodException e) {
+      throw new IllegalArgumentException(
+          String.format("Could not initialize '%s': ", clientBuilderClassName), e);
+    }
+  }
+
+  public Map<String, ContentBloomFilter> getLiveContents(
+      List<Reference> references,
+      Map<String, Instant> cutOffTimestampPerRef,
+      Instant defaultCutOffTimestamp) {
+
+    List<Map<String, ContentBloomFilter>> bloomFilterMaps =
+        new JavaSparkContext(session.sparkContext())
+            .parallelize(references, references.size())
+            .map(
+                reference ->
+                    computeLiveContents(
+                        clientBuilderConfiguration,
+                        cutOffTimestampPerRef,
+                        defaultCutOffTimestamp,
+                        reference))
+            .collect();
+    return mergeLiveContentResults(bloomFilterMaps);
+  }
+
+  public IdentifiedResult getIdentifiedResults(
+      Map<String, ContentBloomFilter> liveContentsBloomFilterMap, List<Reference> references) {
+
+    List<IdentifiedResult> results =
+        new JavaSparkContext(session.sparkContext())
+            .parallelize(references, references.size())
+            .map(
+                reference ->
+                    computeExpiredContents(
+                        clientBuilderConfiguration, liveContentsBloomFilterMap, reference))
+            .collect();
+    // TODO: write each task output directly into the spark table.
```

Comment:
Each executor task can return an iterator of spark row and which can be written directly to the spark table. Instead of accumulating/merging each task output to a single in-memory map.